components/template-mgt/org.wso2.carbon.identity.template.mgt/src/main/java/org/wso2/carbon/identity/template/mgt/TemplateManagerImpl.java

```diff
@@ -302,102 +280,22 @@ public Template getTemplateById(String templateId) throws TemplateManagementExce
-                resourcesList = configManager.getResourcesByType(templateType);
-            } else {
-                resourcesList = configManager.getTenantResources(searchCondition);
-            }
-            return resourcesList.getResources().stream().map(resource -> {
-                resource.setResourceType(templateType);
-                return new ResourceToTemplate().apply(resource);
-            }).collect(Collectors.toList());
-        } catch (ConfigurationManagementException e) {
-            if (ConfigurationConstants.ErrorMessages.ERROR_CODE_RESOURCE_TYPE_DOES_NOT_EXISTS.getCode().equals(e
-                    .getErrorCode())) {
-                if (log.isDebugEnabled()) {
-                    log.debug("Template type : '" + templateType + "' has not been created in the database.", e);
-                }
-                return Collections.emptyList();
-            } else if (ConfigurationConstants.ErrorMessages.ERROR_CODE_RESOURCES_DOES_NOT_EXISTS.getCode().equals(e
-                    .getErrorCode())) {
-                if (log.isDebugEnabled()) {
-                    String message = "Templates do not exist for template type: " + templateType;
-                    if (searchCondition != null) {
-                        message = message + ", and search  criteria:" + searchCondition.toString();
-                    }
-                    log.debug(message, e);
-                }
-                return Collections.emptyList();
-            }
-            throw handleServerException(TemplateMgtConstants.ErrorMessages.ERROR_CODE_LIST_TEMPLATES, e, templateType,
-                    getTenantDomainFromCarbonContext());
-        }
-    }
-
-    @Override
-    public void deleteTemplateById(String templateId) throws TemplateManagementException {
 
-        ConfigurationManager configManager = TemplateManagerDataHolder.getInstance().getConfigurationManager();
-        try {
-            configManager.deleteResourceById(templateId);
-        } catch (ConfigurationManagementException e) {
-            if (ConfigurationConstants.ErrorMessages.ERROR_CODE_RESOURCE_ID_DOES_NOT_EXISTS.getCode().equals(
-                    e.getErrorCode())) {
-                throw handleClientException(TemplateMgtConstants.ErrorMessages.ERROR_CODE_TEMPLATE_NOT_FOUND, e,
-                        templateId, getTenantDomainFromCarbonContext());
-            }
-            throw handleServerException(TemplateMgtConstants.ErrorMessages.ERROR_CODE_DELETE_TEMPLATE_BY_ID, e,
-                    templateId, getTenantDomainFromCarbonContext());
+        List<Template> templates = new ArrayList<>();
+        List<ReadOnlyTemplateHandler> readOnlyTemplateHandlers =
+                TemplateManagerDataHolder.getInstance().getReadOnlyTemplateHandlers();
+        for (ReadOnlyTemplateHandler readOnlyTemplateHandler : readOnlyTemplateHandlers) {
+            templates.addAll(readOnlyTemplateHandler.listTemplates(templateType, limit, offset, searchCondition));
```

Comment:
this might cause problems if we are going to support pagination in the future.. this will paginate from each ReadOnlyTemplateHandler available.. whereas the client expects to paginate the total result from all the handlers.. WDYT?

Reply:
in order to address the above concern, we need to get the totalCount from each template handler and modify the limit to be passed to each handler.. we can address this concern when we support pagination for template API..